hw_7/DOM.js

```diff
@@ -0,0 +1,301 @@
+divShow.querySelector('button ~ button').innerHTML = 'Show Completed';
+
+
+//div .todo
+let divTodo = document.createElement('div');
+divTodo.className = "todo";
+divShow.after(divTodo);
+//ul
+let ul_td = document.createElement('ul');
+divTodo.prepend(ul_td);
+
+// Вариант № 1 (если раскомментировать)
+/*ul_td.innerHTML = getItem('data_myUl');
+
+//li
+function addLi(myText) {
+    ul_td.insertAdjacentHTML('afterbegin', `<li>
+                        <div class="todo-li">
+                            <div>
+                                <input type="checkbox" class="check">
+                                <div><span class="doToText">${myText}</span></div>
+                                <button class="close">x</button>
+                            </div>
+                            <div>
+                                <div><span class="date">${new Date().toLocaleString()}</span></div>
+                            </div>
+                        </div>
+                    </li>`);
+
+    setItem('data_myUl', ul_td.innerHTML);
+}
+
+function test() {
+    alert('Hello');
+}
+
+function onCheck(event) {
+    if (event.target.className === 'check') {
+        test();
+    }
+}
+
+function onAddLi(event) {
+    if (event.target.className === 'add') {
+        let val = document.querySelector('.inputText').value;
+        addLi(val);
+    }
+}
+function onDelLi(event) {
+    if (event.target.className === 'close') {
+        event.target.closest('li').remove();
+    }
+    setItem('data_myUl', ul_td.innerHTML);
+}
+
+function onDelAllLi(event) {
+    if (event.target.className === 'deleteAll') {
+        ul_td.innerHTML = '';
+    }
+    setItem('data_myUl', ul_td.innerHTML);
+}
+
+function onChecked(event) {
+    if (event.target.className === 'check') {
+        event.target.checked ?
+            event.target.nextElementSibling.style.textDecoration = 'line-through' :
+            event.target.nextElementSibling.style.textDecoration = 'none';
+
+    }
+}
+
+function onStor(event) {
+    if (event.key === 'data_myUl') {
+        ul_td.innerHTML = `${event.newValue}`
+    }
+}
+
+function setItem(key, item) {
+    try {
+        localStorage.setItem(key, item);
+    }
+    catch (e) {
+        console.error(e)
+    }
+}
+
+function getItem(key) {
+    try {
+        return localStorage.getItem(key);
+    }
+    catch (e) {
+        console.error(e)
+    }
+}
+
+document.querySelector('.add').addEventListener('click', onAddLi);
+document.querySelector('.deleteAll').addEventListener('click', onDelAllLi);
+document.querySelector('ul').addEventListener('click', onDelLi);
+//document.querySelector('ul').addEventListener('change', onChecked); //onCheck
+window.addEventListener('storage', onStor);
+*/
+
+
+// Вариант № 2
+//ul_td.innerHTML = getItem('data_myUl');
+
+let myLi = [];
+let htmlStr = '';
+let maxId = getMax(myLi);
+
+
+if (getItem('data_myUl')){
+   let myLi = getItem('data_myUl');
+   render(myLi);
+}
```

Comment:
так ок, но минус, что два раза обращаемся к storage
можно так 
```suggestion
let storageItems = getItem('data_myUl');
if (storageItems) {
   let myLi = storageItems;
   render(myLi);
}
```
Ну или сделать через твой любимый тернарный оператор, тут он идеально подходит)
```js
let storageItems = getItem('data_myUl');
let myLi = storageItems ? storageItems : [];
```